[Driver] Fix incorrect GNU triplet for PowerPC on SUSE Linux

On SUSE distributions for 32-bit PowerPC, gcc is configured
as a 64-bit compiler using the GNU triplet "powerpc64-suse-linux",
but invoked with "-m32" by default. Thus, the correct GNU triplet
for 32-bit PowerPC SUSE distributions is "powerpc64-suse-linux"
and not "powerpc-suse-linux".

Differential Revision: https://reviews.llvm.org/D55326

diff -Nru cfe-7.0.0.src.orig/lib/Driver/ToolChains/Gnu.cpp cfe-7.0.0.src/lib/Driver/ToolChains/Gnu.cpp
--- cfe-7.0.0.src.orig/lib/Driver/ToolChains/Gnu.cpp	2018-07-31 16:21:46.000000000 +0200
+++ cfe-7.0.0.src/lib/Driver/ToolChains/Gnu.cpp	2018-12-05 15:55:34.106427845 +0100
@@ -1865,7 +1865,7 @@
   static const char *const PPCLibDirs[] = {"/lib32", "/lib"};
   static const char *const PPCTriples[] = {
       "powerpc-linux-gnu", "powerpc-unknown-linux-gnu", "powerpc-linux-gnuspe",
-      "powerpc-suse-linux", "powerpc-montavista-linuxspe"};
+      "powerpc64-suse-linux", "powerpc-montavista-linuxspe"};
   static const char *const PPC64LibDirs[] = {"/lib64", "/lib"};
   static const char *const PPC64Triples[] = {
       "powerpc64-linux-gnu", "powerpc64-unknown-linux-gnu",
